packages/daemon/src/types.d.ts

```diff
@@ -232,35 +232,41 @@ export type Specials = {
   [specialName: string]: (builtins: Builtins) => Formula;
 };
 
-export type Label = {
-  number: number;
-  who: string;
-  dest: string;
-  when: string;
-  dismissed: Promise<void>;
-};
-export type InternalLabel = Label;
+export interface Responder {
+  respondId(id: string | Promise<string>): void;
+}
 
 export type Request = {
   type: 'request';
-  what: string;
+  description: string;
+  responder: ERef<Responder>;
   settled: Promise<'fulfilled' | 'rejected'>;
 };
-export type InternalRequest = Request;
 
 export type Package = {
   type: 'package';
   strings: Array<string>; // text that appears before, between, and after named formulas.
   names: Array<string>; // edge names
-  formulas: Array<string>; // formula identifiers
+  ids: Array<string>; // formula identifiers
+};
+
+export type Message = Request | Package;
+
+export type EnvelopedMessage = Message & {
+  to: string;
+  from: string;
 };
-export type InternalPackage = Package;
 
-export type Payload = Request | Package;
-export type InternalPayload = InternalRequest | InternalPackage;
+export interface Dismisser {
+  dismiss(): void;
+}
 
-export type Message = Label & Payload;
-export type InternalMessage = InternalLabel & InternalPayload;
+export type StampedMessage = EnvelopedMessage & {
+  number: number;
+  date: string;
+  dismissed: Promise<void>;
+  dismisser: ERef<Dismisser>;
+};
```

Comment:
These are the new degrees of annotation for messages. Message (when proposed) inside an Envelope (on the wire) with a Stamp (in an inbox).